mono/metadata/icall.c

```diff
@@ -3275,6 +3275,28 @@ set_type_object_in_array (MonoDomain *domain, MonoType *type, MonoArrayHandle de
 	HANDLE_FUNCTION_RETURN_VAL (is_ok (error));
 }
 
+void
+ves_icall_RuntimeType_GetGUID (MonoReflectionTypeHandle type_handle, MonoArrayHandle guid_handle, MonoError *error)
+{
+	error_init (error);
+
+	g_assert (mono_array_handle_length (guid_handle) == 16);
+	if (MONO_HANDLE_IS_NULL (type_handle)) {
+		mono_error_set_argument_null (error, "type", "");
+		return;
+	}
+
+	MonoType *type = MONO_HANDLE_GETVAL (type_handle, type);
+	MonoClass *klass = mono_class_from_mono_type_internal (type);
+	if (!mono_class_init_checked (klass, error))
+		return;
+
+	MONO_ENTER_NO_SAFEPOINTS;
+	guint8 *data = (guint8*) mono_array_addr_with_size_internal (MONO_HANDLE_RAW (guid_handle), 1, 0);
+	mono_metadata_get_class_guid (klass, data, error);
+	MONO_EXIT_NO_SAFEPOINTS;
```

Comment:
```suggestion
	guint8 *data = (guint8*) mono_array_addr_with_size_internal (MONO_HANDLE_RAW (guid_handle), 1, 0);
	mono_metadata_get_class_guid (klass, data, error);
```

You don't need the "no safepoints" boundary here.  We recently changed how our local managed object handles work - they prevent the GC from moving the object around.  So you can just assume that `data` will be in the same place until the icall returns.


(and it probably isn't correct, because I think pretty printing the interface type may trigger assembly loading, which may trigger running managed code which could sfepoint).

We recent

Reply:
I updated the commit, removing `MONO_ENTER_NO_SAFEPOINTS`/`MONO_EXIT_NO_SAFEPOINTS`